services/web/server/src/simcore_service_webserver/exporter/formatters/sds/xlsx/xlsx_base.py

```diff
@@ -0,0 +1,292 @@
+import inspect
+from typing import Dict, Generator, Tuple, Any, Set, List, Union
+from pathlib import Path
+from pydantic import BaseModel
+
+from openpyxl import Workbook
+from openpyxl.worksheet.worksheet import Worksheet
+from openpyxl.styles import Border, Alignment
+from openpyxl.cell import Cell
+
+
+def _border_or(self: Border, other: Border) -> Border:
+    return Border(
+        left=self.left or other.left,
+        right=self.right or other.right,
+        top=self.top or other.top,
+        bottom=self.bottom or other.bottom,
+        diagonal=self.diagonal or other.diagonal,
+        diagonal_direction=(self.diagonal_direction or other.diagonal_direction),
```

Comment:
is there any reason for the parenthesis here?

```python
   params = self.__dict__.keys() # assuming this is true, otherwise a list of names
   return Border( **{param: getattr(self, param, None) or getattr(other, param, None)  for param in params }) 
```